src/Symfony/Component/Serializer/Encoder/XmlEncoder.php

```diff
@@ -82,7 +82,12 @@ public function encode(mixed $data, string $format, array $context = []): string
         $encoderIgnoredNodeTypes = $context[self::ENCODER_IGNORED_NODE_TYPES] ?? $this->defaultContext[self::ENCODER_IGNORED_NODE_TYPES];
         $ignorePiNode = \in_array(\XML_PI_NODE, $encoderIgnoredNodeTypes, true);
         if ($data instanceof \DOMDocument) {
-            return $data->saveXML($ignorePiNode ? $data->documentElement : null);
+            set_error_handler(static function ($type, $message) { throw new NotEncodableValueException($message); }, \E_ERROR | \E_WARNING);
```

Comment:
that doesn't work: php won't fallback on the previous handler for deprecation
instead, I suggest using the `@` operator and using error_get_last to compute the exception message when needed

Reply:
```
$result = @$data->saveXML($ignorePiNode ? $data->documentElement : null);

if (false === $result) {
    throw Exception();
}
```
doesn't cover all the case.

For instance
```
$dom = new DOMDocument();

$root = $dom->createElement('root');
$dom->appendChild($root);
$text = $dom->createTextNode("Invalid character: " . chr(7));
$root->appendChild($text);
$saved = $dom->saveXML();

// Check if saving was successful
if ($saved === false) {
    echo "Error saving XML\n";
} else {
    echo "XML saved successfully\n";
    echo $saved;
}
```

gives
```
Warning: DOMDocument::saveXML(): xmlEscapeEntities : char out of range in /home/user/scripts/code.php on line 9
XML saved successfully
<?xml version="1.0"?>
<root></root>
```

Reply:
It should be better with https://github.com/symfony/symfony/pull/54346/commits/41c83a31549de301ae6542f1ba0c3bf4c89b4594 @nicolas-grekas